docs/decisions/0003-django-setting-toggles.rst

```diff
@@ -32,41 +32,29 @@ We will introduce a simple wrapping class (or classes) that can be used to expli
 Here are some example instances::
 
     # Toggle setting
-    EXAMPLE_SETTING_TOGGLE = DjangoSettingToggle("EXAMPLE_TOGGLE", default=False)
+    EXAMPLE_SETTING_TOGGLE = SettingToggle("EXAMPLE_TOGGLE", default=False)
 
     # or
 
-    # Toggle setting in a dict: Option 1
-    EXAMPLE_SETTING_TOGGLE = DjangoSettingToggle(
-        "EXAMPLE_SETTING_DICT['EXAMPLE_TOGGLE']",
-        default=False,
-    )
-
-    # or
-
-    # Toggle setting in a dict: Option 2
-    EXAMPLE_SETTING_TOGGLE = DjangoSettingDictToggle(
+    # Toggle setting in a dict
+    EXAMPLE_SETTING_TOGGLE = SettingDictToggle(
```

Comment:
Could we combine these into one class that has varargs defining a path into a dictionary,? E.g. `SettingToggle("EXAMPLE_TOGGLE", default=False)` and `SettingToggle("EXAMPLE_SETTING_DICT", "EXAMPLE_TOGGLE", default=False)` -- walk down into dicts until the last key, then `get` that one. Or does that end up getting too messy with Python vararg and kwarg declaration?

Reply:
I think the alternative might be:
```
SettingToggle("EXAMPLE_TOGGLE", dict_name="EXAMPLE_SETTING_DICT", default=False)
```
The minor issue with this is that the annotation would read `EXAMPLE_SETTING_DICT['EXAMPLE_TOGGLE']`, which is the opposite order of how you would read the argument names.  I don't feel strongly either way.

Reply:
Yeah, I wouldn't want it to go the other direction.

This is one of those cases where I'd love to be able to have varags _before_ other positional args, which crops up every once in a while in Python and other languages. Like `(*dict_path, setting, default)`.

It wouldn't be terrible to accept `(*setting_path, default=False)` and just pop the last element of `setting_path` off as the actual setting, with the remaining (if any) being the path into a dict. But I don't know if that's too fancy.

Reply:
I feel like the currently documented options are pretty simple to read and understand.

Reply:
FYI: I'm going to leave as-is.